apps/protocol/lib/lexical/protocol/conversions.ex

```diff
@@ -121,31 +117,32 @@ defmodule Lexical.Protocol.Conversions do
         {:ok, LSPosition.new(line: line_count, character: 0)}
 
       true ->
-        with {:ok, line} <- Lines.fetch_line(lines, position.line),
-             {:ok, lsp_character} <- extract_lsp_character(position, line) do
+        with {:ok, lsp_character} <- extract_lsp_character(position) do
           ls_pos =
             LSPosition.new(
               character: lsp_character,
-              line: position.line - lines.starting_index
+              line: position.line - position.starting_index
             )
 
           {:ok, ls_pos}
         end
     end
   end
 
-  def to_lsp(%LSPosition{} = position, _) do
+  def to_lsp(%LSPosition{} = position) do
     {:ok, position}
   end
 
   # Private
 
-  defp extract_lsp_character(%ElixirPosition{} = position, line(ascii?: true, text: text)) do
+  defp extract_lsp_character(
+         %ElixirPosition{context_line: line(ascii?: true, text: text)} = position
+       ) do
     character = min(position.character - 1, byte_size(text))
     {:ok, character}
   end
 
-  defp extract_lsp_character(%ElixirPosition{} = position, line(text: utf8_text)) do
+  defp extract_lsp_character(%ElixirPosition{context_line: line(text: utf8_text)} = position) do
```

Comment:
I think I get it, mainly because the additional information is needed here

Reply:
Yes